syntheseus/tests/reaction_prediction/cli/test_eval.py

```diff
@@ -78,7 +97,43 @@ def get_model_results(remove_duplicates: bool = True, **kwargs):
         assert results_with_repeats == DummyModel.RESULTS
 
 
-def test_print_and_save():
+@pytest.mark.parametrize("forward", [False, True])
+@pytest.mark.parametrize("num_dataset_truncation", [2, 3, 4])
+@pytest.mark.parametrize("num_top_results", [2, 3, 4, 5])
+def test_compute_metrics(
+    forward: bool, num_dataset_truncation: int, num_top_results: int, tmp_path: Path
+) -> None:
+    samples = []
+    for input, output in [("C.N", "CN"), ("NC=O", "CN"), ("C.C", "CC")]:
+        if forward:
+            input, output = output, input
+        samples.append(
+            ReactionSample.from_reaction_smiles_strict(f"{input}>>{output}", mapped=False)
+        )
+
+    for fold in DataFold:
+        DiskReactionDataset.save_samples_to_file(data_dir=tmp_path, fold=fold, samples=samples)
+
+    eval_results = compute_metrics(
+        model=DummyModel(is_forward=forward, repeat=False),
+        dataset=DiskReactionDataset(tmp_path, sample_cls=ReactionSample),
```

Comment:
Do we need to consider other cases of `sample_cls`?

Reply:
I don't think there are any other reaction sample classes in the public code? There are a few in the non-public code though.